lib/purescript-cst/src/Language/PureScript/CST/Types.hs

```diff
@@ -137,6 +137,7 @@ data Type a
   | TypeWildcard a SourceToken
   | TypeHole a (Name Ident)
   | TypeString a SourceToken PSString
+  | TypeInt a SourceToken Integer
```

Comment:
This should be reworked per @JordanMartinez's comment https://github.com/purescript/purescript/pull/4207#issuecomment-980741807.